src/components/articles/ArticlePage.jsx

```diff
@@ -1,11 +1,48 @@
 import React from 'react'
+import { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom'
+import { getArticleById } from '../../api/api'
 
 const ArticlePage = () => {
+  const { article_id } = useParams();
+  const [ article, setArticle] = useState(null)
+  const [isLoading, setIsLoading] = useState(true)
+
+  useEffect(() => {
+    const fetchArticle = async () => {
+      try {
+        setIsLoading(true);
+        const articleData = await getArticleById(article_id);
+        setArticle(articleData.articles[0]);
+      } catch(err) {
+        console.error('Error fetching article', err)
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    fetchArticle();
+  }, [article_id])
+
+  if(isLoading){
+    return <div>Article is loading...</div>
+  }
+
+  if(!article){
+    return <div>Error fetching article!</div>
+  }
```

Comment:
Very nice consideration of loading and error handling! It would be good to use a bit of state for handling the error in much the same way as you use state to handle loading, that could then be used inside the `catch` instead of just console logging 💡